net/frr/src/opnsense/mvc/app/controllers/OPNsense/Quagga/forms/dialogEditBGPNeighbor.xml

```diff
@@ -46,8 +46,8 @@
   <field>
     <id>neighbor.updatesource</id>
     <label>Update-Source Interface</label>
-    <type>dropdown</type>
-    <help><![CDATA[Physical name of the IPv4 interface facing the peer. Please refer to the <a href="http://docs.frrouting.org/en/stable-7.4/bgp.html#clicmd-[no]neighborPEERupdate-source%3CIFNAME|ADDRESS%3E">FRR documentation</a> for more information.]]></help>
+    <type>text</type>
```

Comment:
I thin there is a switch already for having multi protocol support enabled for bgp.
My setup is single protocol but a router instance for v4 and for v6. We could make it standard so that it generally takes the v4 address of the interface and with an opt-in switch to take the v6 address. That way we would support current setups and my the given possibilities/my needs.

That way we could leave it with interface given as selection in a dropdown and add the functionality of having IPv6 as update source.
But then we would have to select an address from the interface given. This is a tricky one.

Reply:
hmm, I probably need some help to understand how this should work for IPv6, the documentation (https://docs.frrouting.org/en/latest/bgp.html) mentions the following:

> Specify the IPv4 source address to use for the BGP session to this neighbour, may be specified as either an IPv4 address directly or as an interface name (in which case the zebra daemon MUST be running in order for bgpd to be able to retrieve interface state).

But if there's a preference for protocol, how should that work then. It does worry me a bit that IPv6 is not mentioned at all (although I might be missing something).